tools/crm_mon.c

```diff
@@ -2009,10 +2010,30 @@ mon_refresh_display(gpointer user_data)
         out->reset(out);
     }
 
-    rc = pcmk__output_cluster_status(out, st, cib, current_cib, fence_history,
-                                     show, show_opts, options.only_node,
-                                     options.only_rsc, options.neg_location_prefix,
-                                     output_format == mon_output_monitor);
+    if ((cib != NULL) && (cib->state != cib_disconnected)) {
+        rc = pcmk__output_cluster_status(out, st, cib, current_cib,
+                                         fence_history, show, show_opts,
+                                         options.only_node, options.only_rsc,
+                                         options.neg_location_prefix,
+                                         output_format == mon_output_monitor);
+    } else {
+        exit_code = CRM_EX_DISCONNECT;
+
+        // Print an unavailable message for daemonized crm_mon
+        switch (output_format) {
```

Comment:
We **could** be create formatter functions that simply print "Cluster is not available" in the right way for each format, and call `out->message()` here. That would be a bit more involved than this approach (and would require adding an attribute or element to the XML schema), but it's certainly doable.

Reply:
Definitely, we should never switch on format to choose what display